src/expertsystem/amplitude/sympy.py

```diff
@@ -0,0 +1,428 @@
+"""Generate an amplitude model in terms of SymPy objects."""
+
+import logging
+import operator
+from functools import reduce
+from typing import Dict, Generic, List, Optional, Tuple, TypeVar
+
+import attr
+import sympy as sy
+from sympy.abc import x
+from sympy.physics.quantum.dagger import Dagger
+from sympy.physics.quantum.spin import Rotation as Wigner
+
+from expertsystem.particle import Particle, ParticleCollection
+from expertsystem.reaction import Result
+from expertsystem.reaction.combinatorics import (
+    perform_external_edge_identical_particle_combinatorics,
+)
+from expertsystem.reaction.quantum_numbers import ParticleWithSpin
+from expertsystem.reaction.topology import StateTransitionGraph
+
+from ._graph_info import (
+    generate_kinematics,
+    generate_particle_collection,
+    get_prefactor,
+    group_graphs_same_initial_and_final,
+)
+from .model import Kinematics
+
+ValueType = TypeVar("ValueType", float, complex)
+
+
+@attr.s(auto_attribs=True)
+class _HelicityParticle:
+    particle: Particle
+    helicity: float
+
+
+@attr.s
+class InitialValue(Generic[ValueType]):
+    value: ValueType = attr.ib()
+    fix: bool = attr.ib()
+
+
+@attr.s
+class SympyModel:
+    kinematics: Kinematics = attr.ib(
+        validator=attr.validators.instance_of(Kinematics)
+    )
+    particles: ParticleCollection = attr.ib(
+        validator=attr.validators.instance_of(ParticleCollection)
+    )
+    intensity_model: sy.Expr = attr.ib(default=None)
+    components: Dict[sy.Function, sy.Function] = attr.ib(default=dict())
+    dynamics: Dict[sy.Function, sy.Function] = attr.ib(default=dict())
+    initial_values: Dict[sy.Symbol, InitialValue] = attr.ib(default=dict())
+
+
+class _SympyHelicityAmplitudeNameGenerator:
```

Comment:
I did not fully analyze this class, but it seems to be very similar to the existing normal helicity amplitude name generator `_HelicityAmplitudeNameGenerator`.

Is the idea here to just remove the other file once the sympy code has been verified or why was this class/functionality copied?

Reply:
True, this class just started out as a copy of `_HelicityAmplitudeNameGenerator`. The main thing that's different here is that `Particle.latex` is used if available, because then the symbol names render more nicely in the notebooks.

We may have to think about a better naming scheme though... currently the symbol names have become horribly long. So either we provide a mapping from numbered symbols (e.g. `"C_1": "C[J/\\psi(1S) \\to f_{0}(980)_{0} \\gamma_{+1};f_{0}(980) \\to \\pi^{0}_{0} \\pi^{0}_{0}]]'))"`). Or we would have to provide some mapping that tells what those indices mean.

Reply:
A mapping would be most flexible, basically we should always think in terms of that module not in terms of what the user would like to see at the end. So in other words design the dream interface we wish to have for that module and write some adapters or converters for it